huaweicloud/services/live/resource_huaweicloud_live_record_callback.go

```diff
@@ -0,0 +1,228 @@
+package live
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	v1 "github.com/huaweicloud/huaweicloud-sdk-go-v3/services/live/v1"
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/live/v1/model"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+const (
+	eventRecordNewFileStart = "RECORD_NEW_FILE_START"
+	eventRecordFileComplete = "RECORD_FILE_COMPLETE"
+	eventRecordOver         = "RECORD_OVER"
+	eventRecordFailed       = "RECORD_FAILED"
+
+	allAppName = "*"
+)
+
+func ResourceRecordCallback() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceRecordCallbackCreate,
+		ReadContext:   resourceRecordCallbackRead,
+		UpdateContext: resourceRecordCallbackUpdate,
+		DeleteContext: resourceRecordCallbackDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"domain_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"url": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^http://|^https://`),
+						"The URL must start with http:// or https://."),
+					validation.StringDoesNotMatch(regexp.MustCompile(`\?{1}`), "The URL cannot contain parameters."),
+				),
+			},
+
+			"type": {
```

Comment:
suggest to use `types` instead as it's a list

Reply:
done